addons/website_sale_loyalty/tests/test_website_sale_auto_invoice.py

```diff
@@ -0,0 +1,43 @@
+from odoo.tests import tagged
+from odoo.fields import Command
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo.addons.website.tools import MockRequest
+from odoo.addons.website_sale.tests.common import WebsiteSaleCommon
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteSaleAutoInvoice(WebsiteSaleCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.Controller = WebsiteSale()
+
+    def test_automatic_invoice_on_zero_amount(self):
+        # Set automatic invoice
+        self.env['ir.config_parameter'].sudo().set_param('sale.automatic_invoice', 'True')
+
+        # Create a discount code
+        program = self.env['loyalty.program'].sudo().create({
+            'name': '100discount',
+            'program_type': 'promo_code',
+            'rule_ids': [
+                Command.create({
+                    'code': "100code",
+                    'minimum_amount': 0,
+                })
+            ],
+            }
+        )
+        reward = self.env['loyalty.reward'].sudo().create({
+            'program_id': program.id,
+            'discount': 100,
+        })
```

Comment:
```suggestion
        program = self.env['loyalty.program'].sudo().create({
            'name': '100discount',
            'program_type': 'promo_code',
            'rule_ids': [
                Command.create({
                    'code': "100code",
                    'minimum_amount': 0,
                }),
            ],
            'reward_ids': [
                Command.create({...}
            ],
            }
        )
        reward = program.reward_ids
```
When possible, create the records directly with the complete data, it's more efficient.
I believe that in this case, the program would create a reward automatically, and end up with two rewards, which seems unnecessary for the purpose of your test.